datadog_lambda/tracing.py

```diff
@@ -377,30 +379,49 @@ def extract_context_from_step_functions(event, lambda_context):
     into lambda's event dict.
     """
     try:
-        execution_id = event.get("Execution").get("Id")
-        state_name = event.get("State").get("Name")
-        state_entered_time = event.get("State").get("EnteredTime")
-        # returning 128 bits since 128bit traceId will be break up into
-        # traditional traceId and _dd.p.tid tag
-        # https://github.com/DataDog/dd-trace-py/blob/3e34d21cb9b5e1916e549047158cb119317b96ab/ddtrace/propagation/http.py#L232-L240
-        trace_id = _deterministic_sha256_hash(execution_id, LOWER_64_BITS)
-
-        parent_id = _deterministic_sha256_hash(
-            f"{execution_id}#{state_name}#{state_entered_time}", HIGHER_64_BITS
-        )
+        meta = {}
+
+        if "_datadog" in event:
+            dd_data = event.get("_datadog")
+            parent_id = _sha256_to_binary_part(
+                dd_data.get("x-datadog-parent-id-hash"), HIGHER_64_BITS
```

Comment:
Note: Since this `extract_context_from_step_functions()` is executed when we know `_datadog.serverless-version=v2`, we don't need to check if this hash key exists. The v2 contract is that they need to have the `_datadog.x-datadog-parent-id-hash`.

Reply:
I might be misunderstanding, we're not checking if the hash key exists right?

We're computing the `parent_id` no matter what, the only thing we check in this branch is if `x-datadog-trace-id` exists

Reply:
So we're saying it's the same `EventTypes.STEP_FUNCTIONS` whether its the old format with context object or new one with `_datadog`. Both cases will enter `extract_context_from_step_functions()`

If we want to separate them, one thing we could do is make a new V2 event with a V2 extractor to go along with it